pkg/restore/restore.go

```diff
@@ -634,20 +636,73 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 		}
 
 		// necessary because we may have remapped the namespace
-		obj.SetNamespace(namespace)
+		// if the namespace is blank, don't create the key
+		if namespace != "" {
+			obj.SetNamespace(namespace)
+		}
 
 		// add an ark-restore label to each resource for easy ID
 		addLabel(obj, api.RestoreLabelKey, ctx.restore.Name)
 
 		ctx.infof("Restoring %s: %v", obj.GroupVersionKind().Kind, obj.GetName())
-		_, err = resourceClient.Create(obj)
-		if apierrors.IsAlreadyExists(err) {
-			addToResult(&warnings, namespace, err)
-			continue
+		// Use a special error since we need to retain that information through a couple checks.
+		_, restoreErr := resourceClient.Create(obj)
+		if apierrors.IsAlreadyExists(restoreErr) {
+			fromCluster, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{})
+			if err != nil {
+				addToResult(&warnings, namespace, err)
+				continue
+			}
+			// Remove insubstantial metadata
+			fromCluster, err = resetMetadataAndStatus(fromCluster)
+			if err != nil {
+				addToResult(&warnings, namespace, err)
+				continue
+			}
+			if obj.GroupVersionKind().Kind == "ServiceAccount" {
```

Comment:
TODO: make this the type instead of the string name

Reply:
Extract this to a helper function. That way you don't have to do a ton of `continue`s, plus it's easier to test.